### ISSUE TITLE
vak.Model does not save checkpoint at end of training

### ISSUE THREAD

**Issue by NickleDave** [NickleDave/vak#157]
Currently a checkpoint only gets saved when:
1. validation accuracy increases after an epoch, or
2. it is a checkpoint epoch, i.e. `epoch % checkpoint_step == 0`

No checkpoint gets saved *after* the training loop ends.
But most users will want a checkpoint to get saved after the last epoch--why else would they train that many epochs?
in the extreme case, if training is only one epoch, and `checkpoint step` > 1, then no checkpoint ever gets saved.
